diff -up binaryen-version_95/scripts/test/wasm2js.py.node binaryen-version_95/scripts/test/wasm2js.py
--- binaryen-version_95/scripts/test/wasm2js.py.node	2020-07-21 06:04:33.000000000 +0200
+++ binaryen-version_95/scripts/test/wasm2js.py	2020-07-21 13:43:50.093293525 +0200
@@ -80,7 +80,7 @@ def test_wasm2js_output():
                 # `spectest` and `env` modules in our tests.
                 if shared.NODEJS:
                     loader = os.path.join(shared.options.binaryen_root, 'scripts', 'test', 'node-esm-loader.mjs')
-                    node = [shared.NODEJS, '--experimental-modules', '--loader', loader]
+                    node = [shared.NODEJS, '--experimental-modules', '--no-warnings', '--loader', loader]
                     cmd = node[:]
                     cmd.append('a.2asm.mjs')
                     out = support.run_command(cmd)
